core/divelist.c

```diff
@@ -1737,6 +1738,25 @@ bool dive_less_than(const struct dive *a, const struct dive *b)
 	return comp_dives(a, b) < 0;
 }
 
+/* Trips are compared according to the first dive in the trip.
+ * Even though it shouldn't happen, take care about "empty" trips.
+ * Since a dive can only belong to one trip, no two trips should
+ * compare as equal
+ */
+static int comp_trips(const struct dive_trip *a, const struct dive_trip *b)
+{
+	if (a->dives.nr <= 0)
```

Comment:
when would the number be negative?

Reply:
Hopefully never. Actually it should also never be zero - this is just "defensive" coding so I thought lets catch all the bogus cases. But I can change it to `== 0` if you want.

Reply:
I'm ok with defensive coding. I love a comment that says "should never happen, but heck, just to be safe" or something...